### PR TITLE
Add rpm-comps-upload support.

### LINKED ISSUE #406: Add support for comps.xml uploads to rpm
## Summary

PR https://github.com/pulp/pulp_rpm/pull/2159 adds support for uploading comps.xml files. Add support for same to cli
## Examples

`pulp rpm content comps upload --file --repository --replace`

### PR DESCRIPTION
**Pull request by ggainey**
closes #406
Required PR: https://github.com/pulp/pulp_rpm/pull/2159

### CODE REVIEW

**Comment by mdellweg** [comment]
I see the RPM PR is still in flight. Should we draft this?

File: pulpcore/cli/rpm/context.py
Comment (mdellweg) [inline pulpcore/cli/rpm/context.py:39]:
```suggestion
    UPLOAD_COMPS_ID: ClassVar[str] = "upload_comps"
```
Is there any reason not to call this `UPLOAD_ID`?

**Comment by ggainey** [comment]
> I see the RPM PR is still in flight. Should we draft this?

Prob a good idea, I still working to get a response from the stakeholder that the API in 2159 meets their needs.

File: pulpcore/cli/rpm/context.py
Comment (ggainey) [reply pulpcore/cli/rpm/context.py:39]: > Is there any reason not to call this `UPLOAD_ID`?

Mmm, I believe the original intent was to differentiate from the uploading of model-entities like Package and Advisory. Prob not useful, will change.

Comment (mdellweg) [reply pulpcore/cli/rpm/context.py:39]: Actually, i kind of think it should be `rpm_comps_upload` on the server side...

Comment (ggainey) [reply pulpcore/cli/rpm/context.py:39]: Hrm, that was my initial id, but then I changed it to conform to copy - I should have stuck w/ my first thought, I suppose. Well, the PR is waiting for final stakeholder approval, so we can fixit!

**Comment by ggainey** [comment]
@mdellweg ready for final review

File: pulpcore/cli/rpm/context.py
Comment (mdellweg) [inline pulpcore/cli/rpm/context.py:48]: Are we quite confident, these never need to go through the uploads api for being too big?
(From the CLI view, i kind of think, there should only have been a single way to upload data into pulp, but at least every other serializer that accepts files, should also accept uploads or artifacts.)

File: pulpcore/cli/rpm/comps.py
Comment (mdellweg) [inline pulpcore/cli/rpm/comps.py:37]: You should turn this into a `resource_option`.

File: pulpcore/cli/rpm/comps.py
Comment (mdellweg) [inline pulpcore/cli/rpm/comps.py:24]: With the resource option, we don't do this anymore.

File: pulpcore/cli/rpm/context.py
Comment (ggainey) [reply pulpcore/cli/rpm/context.py:48]: comps.xml isn't recognized by pulp_rpm as an entity itself - it's a convenient way of getting the PackageCatalog/Environment/Langpack/Groups uploaded, but there is no 'comps.xml' file anywhere. As far as size goes, the largest comps.xml I know of is Fedora-Updates; F34 unzipped is 1.6Mb. That's def Large; you're right, maybe I need to rethink "it's just a file".

File: pulpcore/cli/rpm/comps.py
Comment (ggainey) [reply pulpcore/cli/rpm/comps.py]: :+1:

File: pulpcore/cli/rpm/comps.py
Comment (mdellweg) [reply pulpcore/cli/rpm/comps.py:36]: I meant the whole "-t" option seems superfluous at this point. Also as you just explained, the "comps" is not a "thing" in Pulp. So i would be ok to say we can merge the command in to "`comps-upload`". WDYT?

File: pulpcore/cli/rpm/comps.py
Comment (mdellweg) [inline pulpcore/cli/rpm/comps.py:60]: This is probably wrong. Keep in mind that this docstring is user facing.

File: pulpcore/cli/rpm/context.py
Comment (ggainey) [reply pulpcore/cli/rpm/context.py:48]: The chunked-upload-api doesn't play very well with either "convenience API" or "not-an-artifact". Given that the largest comps.xml I know if is 1.7Mb, comps.xml isn't its own artifact, and the pulp_rpm create calls all have --file= parameters, I think this is the way to go.

If at some point we decide to, say, add a comps-Artifact and extends the POST /comps/ call  to include an --artifact option, we can extend the CLI to handle it using chunked uploads. But for the purposes of filling the RFE, I think this is sufficient, and consistent.

File: pulpcore/cli/rpm/comps.py
Comment (ggainey) [reply pulpcore/cli/rpm/comps.py:60]: Ooo, nice catch - copypasta strikes again!

File: pulpcore/cli/rpm/comps.py
Comment (ggainey) [reply pulpcore/cli/rpm/comps.py]: I think this all sounds like a fine idea. Ready for review.

**Review by mdellweg — APPROVED** [review]

**Pull request merged**
